vaadin-text-field-flow-integration-tests/src/test/java/com/vaadin/flow/component/textfield/tests/BigDecimalFieldPageIT.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright 2000-2019 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.component.textfield.tests;
+
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+
+import com.vaadin.flow.component.textfield.BigDecimalField;
+import com.vaadin.flow.component.textfield.testbench.BigDecimalFieldElement;
+import com.vaadin.flow.testutil.AbstractComponentIT;
+import com.vaadin.flow.testutil.TestPath;
+import com.vaadin.testbench.TestBenchElement;
+
+import static org.junit.Assert.assertFalse;
+import static org.openqa.selenium.support.ui.ExpectedConditions.attributeToBe;
+
+/**
+ * Integration tests for {@link BigDecimalField}.
+ */
+@TestPath("big-decimal-field-test")
+public class BigDecimalFieldPageIT extends AbstractComponentIT {
```

Comment:
I tried to test also the invalid input prevention, but couldn't make it work. First I tried with `sendKeys`, and then by firing the `keydown` events with JS:
```java
    @Test
    public void preventsInvalidCharacters() {
        String validChars = "1234567890+-.";
        String invalidChars = "aAeE,_/*<>'^";

        Arrays.asList(validChars.split(""))
                .forEach(key -> assertKeyDownPrevented(key, false));
        Arrays.asList(invalidChars.split(""))
                .forEach(key -> assertKeyDownPrevented(key, true));
    }

    private void assertKeyDownPrevented(String key, boolean expectedPrevented) {
        boolean defaultPrevented = (boolean) executeScript(
                "const event = new CustomEvent('keydown',"
                        + "{bubbles: true, cancelable: true, composed: true});"
                        + "event.key = arguments[0];"
                        + "arguments[1].dispatchEvent(event);"
                        + "return event.defaultPrevented;",
                key, field);

        String errorMessage = String.format("Expected key %s to %s.", key,
                expectedPrevented ? "be prevented" : "not be prevented");

        Assert.assertEquals(errorMessage, expectedPrevented, defaultPrevented);
    }
```
The same script works perfectly fine when I run it in Chrome dev tools, but in this test environment `event.defaultPrevented` always returns false.

Anyway, it might not be mandatory to test here, as it's covered (with a different pattern) by `<vaadin-integer-field>` tests.